Dockerfile

```diff
@@ -0,0 +1,76 @@
+### Build stage ###
+FROM golang:1.12-alpine AS build
+
+RUN \
+echo "**** install build dependencies ****" && \
+apk add --no-cache gcc git musl-dev && \
+mkdir /app
+
+WORKDIR /app
+
+# When building from the gohub repository, just copy it in
+copy . .
+# When building from outside the gohub repository, grab the source
+#RUN git clone --depth=1 https://github.com/direct-connect/go-dcpp.git .
+
+# Get dependencies
+RUN \
+echo "**** Install docker start script ****" && \
+mv -v start-docker.sh /bin/go-dcpp && \
+echo "**** get go dependencies ****" && \
+go get -d -v ./... && \
+echo "**** build and install GoHub ****" && \
+go install -v ./...
+
+# Whether to run the go-dcpp tests (0 = no, 1 = yes)
+ARG RUN_TESTS
+ENV RUN_TESTS=${RUN_TESTS}
+
+# The tests are currently flaky, so skip them by default
+RUN \
+if [ "${RUN_TESTS:-0}" = "1" ]; then \
+echo "**** Testing GoHub ****" && \
+go test ./...; \
+else \
+echo "**** Skipping GoHub Tests ****"; \
+fi
+
+
+### Final container stage ###
+FROM alpine:latest
```

Comment:
It's better to pin a specific version here. Sometimes Alpine updated Alpine images will fail to run Go binaries because those were build with a different musl version. It's probably a good idea to pin alpine version in the Go builder image too.